### PR TITLE
Issue-5147 Add app store icon on iOS

### LINKED ISSUE #5140: Cleanup of old app store icons in game.project
**Task (REQUIRED):**
We have fields for app store icons for iOS 6.1 or earlier. Defold minimum iOS version is 8.0.

**Expected outcome (REQUIRED):**
Old icon size fields should no longer be shown in game.project

### LINKED ISSUE #5147: Add support for App Store icon
**Is your feature request related to a problem? Please describe (REQUIRED):**
Uploads to iTunes Connect requires a 1024x icon for the App Store listing present to be able to submit for review

**Describe the solution you'd like (REQUIRED):**
It should be possible to somehow provide a 1024x icon when bundling

**Describe alternatives you've considered (REQUIRED):**
Manually modify the IPA after it was built

**Additional context (OPTIONAL):**
https://forum.defold.com/t/1024-x-1024-icon-asset-missing-for-ios-solved/65592/22

### PR DESCRIPTION
**Pull request by britzl**
Fixes #5147

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java [hunk 1]
```diff
@@ -604,19 +604,15 @@ private BufferedImage getFallbackIconImage(String categoryName, String[] alterna
     public void copyIosIcons() throws IOException
     {
         // Get fallback icon as an image
-        String[] iconPropNames = { "app_icon_57x57", "app_icon_72x72", "app_icon_76x76", "app_icon_114x114", "app_icon_120x120", "app_icon_144x144", "app_icon_152x152", "app_icon_167x167", "app_icon_180x180" };
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java:607]: Icons for iOS 6.1 or older

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java [hunk 2]
```diff
@@ -260,12 +260,34 @@ public void bundleApplication(Project project, File bundleDir, ICanceled cancele
                     throw e;
                 }
             }
-
         } else
         {
             logger.log(Level.WARNING, "ios.launch_screen is not set");
         }
 
+        String icons = projectProperties.getStringValue("ios", "icons");
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java:268]: Copy the .car file containing the App Store icons.

File: engine/engine/content/builtins/manifests/ios/Info.plist [hunk 3]
```diff
@@ -3,40 +3,24 @@
...
-                <string>Icon@2x.png</string>
-                <string>Icon.png</string>
-                <string>Icon-72@2x.png</string>
-                <string>Icon-72.png</string>
-        </array>
         <key>CFBundleIcons</key>
         <dict>
                 <key>CFBundlePrimaryIcon</key>
                 <dict>
+                        <key>CFBundleIconName</key>
```
Comment (britzl) [inline engine/engine/content/builtins/manifests/ios/Info.plist:17]: This references the entry in Asset.car

File: engine/engine/content/builtins/manifests/ios/Info.plist [hunk 4]
```diff
@@ -3,40 +3,24 @@
 <plist version="1.0">
 <dict>
         <key>BuildMachineOSBuild</key>
-        <string>13A603</string>
+        <string>19G73</string>
```
Comment (britzl) [inline engine/engine/content/builtins/manifests/ios/Info.plist:6]: This had to come up

File: engine/engine/content/builtins/manifests/ios/Info.plist [hunk 5]
```diff
@@ -63,11 +47,11 @@
         <key>DTCompiler</key>
         <string>com.apple.compilers.llvm.clang.1_0</string>
         <key>DTPlatformBuild</key>
-        <string>16B91</string>
+        <string>17G64</string>
```
Comment (britzl) [inline engine/engine/content/builtins/manifests/ios/Info.plist:50]: And this had to be increased

File: engine/engine/content/builtins/manifests/ios/Info.plist [hunk 6]
```diff
@@ -63,11 +47,11 @@
...
         <key>DTPlatformName</key>
         <string>iphoneos</string>
         <key>DTPlatformVersion</key>
-        <string>8.0</string>
+        <string>12.1</string>
```
Comment (britzl) [inline engine/engine/content/builtins/manifests/ios/Info.plist:54]: And this

**Comment by britzl** [comment]
Documentation update: https://github.com/defold/doc/blob/master/docs/en/manuals/ios.md#creating-an-icon-asset-catalog

File: engine/engine/content/builtins/manifests/ios/Info.plist [hunk 7]
```diff
@@ -3,40 +3,24 @@
...
         <key>BuildMachineOSBuild</key>
-        <string>13A603</string>
+        <string>19G73</string>
         <key>CFBundleDevelopmentRegion</key>
         <string>{{ios.default_language}}</string>
         <key>CFBundleDisplayName</key>
         <string>{{project.title}}</string>
         <key>CFBundleExecutable</key>
         <string>{{exe-name}}</string>
-        <key>CFBundleIconFiles</key>
```
Comment (britzl) [inline engine/engine/content/builtins/manifests/ios/Info.plist:13]: CFBundleIcons (from iOS 5.0) takes precedence over CFBundleIconFiles if both are present.

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/meta.properties [hunk 8]
```diff
@@ -319,23 +319,18 @@ max_count.default = 128
...
-app_icon_114x114.type = resource
-app_icon_114x114.help = retina iPhone/iPod touch bundle icon file (.png) [required for iOS 6.1 and earlier]
-app_icon_72x72.type = resource
-app_icon_72x72.help = iPad icon file (.png) [required for iOS 6.1 and earlier]
-app_icon_144x144.type = resource
-app_icon_144x144.help = retina iPad bundle icon file (.png) [required for iOS 6.1 and earlier]
+app_icon_152x152.help = icon for home screen on iPad with retina display (.png)
 app_icon_167x167.type = resource
-app_icon_167x167.help = iPad Pro bundle icon file (.png)
+app_icon_167x167.help = icon for ome screen on iPad Pro (.png)
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/meta.properties:330]: "ome" -> "home" ?

File: editor/resources/meta.edn [hunk 9]
```diff
@@ -399,51 +399,33 @@
...
-   :path ["ios" "app_icon_120x120"]}
+   :help "icon for home screen on iPad with retina display (.png)",
+   :path ["ios" "app_icon_152x152"]}
   {:type :resource,
    :filter "png",
    :preserve-extension true,
-   :help
-   "retina iPad bundle icon file (.png) [required for iOS 6.1 and earlier]",
-   :path ["ios" "app_icon_144x144"]}
+   :help "icon for ome screen on iPad Pro (.png)",
```
Comment (JCash) [inline editor/resources/meta.edn:412]: "ome" -> "home"

**Review by JCash — DISMISSED** [review]

**Pull request merged**